onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -2338,6 +2338,15 @@ ORT_API(const OrtTrainingApi*, OrtApis::GetTrainingApi, uint32_t version) {
 #endif
 }
 
+ORT_API(bool, OrtApis::IsTrainingApiAvailable) {
```

Comment:
We discussed on teams, just wanted to summarize here.

Looks like GetTrainingApi will only fail if it's not the training build as 'version' is unused. So what we can do is update the documentation to say that 'version' must be ORT_API_VERSION and we'll only write an error if 'version' isn't valid.

Otherwise it'll return nullptr without error.